### PR TITLE
MULE 18398: Grizzly SSLFilter at write operation is not validating handshake status before call beginHandshake

### PR DESCRIPTION
**Pull request by mlischetti**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -17,10 +17,11 @@
     <version>1.6.0-SNAPSHOT</version>
 
     <properties>
-        <asyncHttpClientVersion>1.14-MULE-018-SNAPSHOT</asyncHttpClientVersion>
-        <grizzlyVersion>2.3.36-MULE-018</grizzlyVersion>
+        <asyncHttpClientVersion>1.14-MULE-019-SNAPSHOT</asyncHttpClientVersion>
```
Comment (eze210) [inline pom.xml:20]: Use released version

File: src/test/java/org/mule/service/http/impl/functional/client/HttpClientTlsCustomProviderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import io.qameta.allure.Story;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.jsse.provider.BouncyCastleJsseProvider;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.Collection;
+
+@Story(TLS)
```
Comment (eze210) [inline src/test/java/org/mule/service/http/impl/functional/client/HttpClientTlsCustomProviderTestCase.java:51]: Add @Feature and @Issue Allure annotations

File: pom.xml (hunk 1 above)
Comment (mlischetti) [reply pom.xml]: Done!

File: src/test/java/org/mule/service/http/impl/functional/client/HttpClientTlsCustomProviderTestCase.java (hunk 2 above)
Comment (mlischetti) [reply src/test/java/org/mule/service/http/impl/functional/client/HttpClientTlsCustomProviderTestCase.java:53]: Added @issue but @feature was defined at superclass

**Comment by mlischetti** [comment]
--validate

**Pull request merged**